smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -198,6 +202,10 @@ private void parseControlSection() {
 
             if (key.equals("version")) {
                 onVersion(value);
+            } else if (key.equals("operationInputSuffix")) {
+                operationInputSuffix = value.expectStringNode().getValue();
```

Comment:
How much do we care about dupes in a single file? Seems like it would be silly for someone to duplicate these keys right, since they have to be at the top of the file... so guarding against it is probably going overboard.

Guarding against multiple versions per/file is still reasonable IMO because it's going to be used with low-level tools like grep that don't know Smithy model semantics. But these inline things probably won't be.

Reply:
I don't think it matters that much to be honest, but I can add some generic check to make sure no control statement is duplicated.